### DISCUSSION TITLE
RNBootSplash with react-native-auth0 not working on Android?

### DISCUSSION THREAD

**Discussion by VilleMiekkoja** [zoontek/react-native-bootsplash#530] [Q&A]
I have a react-native project with react-native-bootsplash. I'm using `useEffect` for hiding the screen like this:

```tsx
const IntroScreen: React.FC<IntroScreenProps> = ({
  navigation,
}) => {
  useEffect(() => {
    RNBootSplash.isVisible().then(isVisible => {
      if (isVisible) {
        return RNBootSplash.hide({ fade: false });
      }
    });
  }, []);

  return <Text>Is this visible?</Text>;
};
```

The setup is so that when user arrives to this page, the splash-screen has already been hidden before in the previous page. From the previous page we move to this page, and then the splashscresn stays on (without logo, just the splashscreen background color on full screen). There's also auth0 login before we move from the previous page (that is a login page). Not sure if it affects to this. But if I comment the whole useEffect out, then there's no issue. But why do I then want to try hiding it in this page if it has already been hidden in the previous page? That is because in some cases this page might be the first page already (if user has logged in before). I could of course maybe send some kind of variable to this page that if we already know the bootsplash has been hidden, but I see no reason why the code above should not work on this situation as well. The issue only happens on android. On iOS it's working fine.

I will setup a workaround for this unless someone knows what I'm doing wrong? Is it not allowed to call isVisible() function if it has already been called once? I fear this might be some kind of timing issue. Here are the relevant library versions:

"@react-navigation/native": "^6.1.7",
"@react-navigation/native-stack": "^6.9.13",
"react-native": "0.72.3",
"react-native-bootsplash": "^5.1.3",

Android emulator: Pixel 2 API 33

**Comment by zoontek** [comment]
@VilleMiekkoja There no need to call `isVisible` before `hide`. If it's already hidden, nothing will happen and the promise will resolve.

```tsx
const IntroScreen: React.FC<IntroScreenProps> = ({
  navigation,
}) => {
  useEffect(() => {
    RNBootSplash.hide({ fade: false });
  }, []);

  return <Text>Is this visible?</Text>;
};
```

**Reply by VilleMiekkoja** [reply]
Unfortunately, even calling the RNBootSplash.hide seems to cause the same stuck on splash screen background =/. Not sure what is happening. Could it be that once we come back from auth0 browser authentication, then the bootsplash on android is initialised again in some way. Because I don't even see the page we are returning to. When it works we return first to the page we left, and it takes some small time before we go to the new page. But if I just put empty useEffect to the new page like this:

```
  useEffect(() => {
    console.log('something')
  }, []);
  ```
  Then it works ok.

**Reply by zoontek** [reply]
Not sure this is related to the library, but probably to your flow. You should try to isolate the issue by creating a minimal reproduction example.

**Reply by VilleMiekkoja** [reply]
Creating a fully reproduction example might be tedious, as it would require setting the auth0 browser authentication. But let's see I might make it. I've now further debugged the issue. It seems if I do anything async inside the useEffect, the same problem occurs. So this will also make the splash-screen background to appear:

```
  useEffect(() => {
    Promise.resolve();
  }, []);
```

While this won't:

```
  useEffect(() => {
    console.log('something');
  }, []);
```

**Reply by zoontek** [reply]
At least having a video would help.

But I doubt of something: you mention background, I'm wondering myself if you are not in this case: https://github.com/zoontek/react-native-bootsplash/issues/516

The solution is to move the theme from the Application to the MainActivity. I updated the README at this time.

**Reply by VilleMiekkoja** [reply]
Thx, I had not up to date configurations in the AndroidManifest.xml. I initialised the library already in Septemper of 2023... Unfortunately the change did not resolve the issue =/. But it did change something. Instead of me now seeing an empty screen with the color of the bootsplash background, I see just a white blank screen. I'll try to add a video of it.

**Reply by VilleMiekkoja** [reply]
https://github.com/zoontek/react-native-bootsplash/assets/2777633/dd0b0785-fe34-4f34-abc4-417acfd052e3

**Reply by zoontek** [reply]
Yeah, so not related. It goes back to your app, but for some reason your app RN app isn't mounted anymore at this point.

You thought it was related to this library because it seems to mount a new Activity, and as the bootsplash theme was the default for your all application… you saw it. That's why trying to reproduce it, starting with a minimal project, the latest version and the documentation is always good.

**Reply by VilleMiekkoja** [reply]
The screens are in the back, the console logs printed in useEffect of the screen behind still show up even the screen does not show because there is some layer on top of it.

**Reply by zoontek** [reply]
Maybe your authentication module doesn't close all the Activities it open?

**Reply by VilleMiekkoja** [reply]
Could be. Next I will try the login with bootsplash completely removed and see if the problem still occurs. You might be very right that this is not related at all to the splash-screen

**Reply by VilleMiekkoja** [reply]
Tested by removing bootsplash. Then the app is working just fine on Android. User comes back from login successfully and is directed to the correct page. This indicates that at minimum the bootsplash probably has some kind of role in this bug.

**Reply by zoontek** [reply]
You're sure that you can't create a small repro? I can create an Auth0 project, account etc on my own, but if I don't have your code / flow I might not be able to reproduce it.

**Reply by VilleMiekkoja** [reply]
Wondering if it would be just easier to give you access rights to the repository. I fear the issue might be some kind of race condition, or memory related. Why would it otherwise make a difference what I put to the `useEffect ` of the next screen that we move after login? I'll test if I can reproduce the error also against a cloud backend. That way running the app locally would be enough and would not need to think about how you would launch the backend part etc..

**Reply by VilleMiekkoja** [reply]
Was able to reproduce it against production backend also. I can give you read access to the repository and guide how to test this @zoontek . I have it on bitbucket. Is this your email: zoontek(at)gmail.com?

**Reply by zoontek** [reply]
Yes, this is my email 🙂

**Reply by VilleMiekkoja** [reply]
I sent you an invite.

**Reply by ppablo28** [reply]
Hello, @VilleMiekkoja did you find a solution for this? In our case removing this code 
https://github.com/zoontek/react-native-bootsplash/blob/master/android/src/main/java/com/zoontek/rnbootsplash/RNBootSplashModuleImpl.java#L106 
solves the issue, but looks like an important one...

**Reply by ppablo28** [reply]
Or @zoontek maybe you could explain how to solve this? I would be very grateful.
